options.go

```diff
@@ -55,6 +55,39 @@ type options struct {
 	gCHeapOpts *typeOption
 	cpuOpts    *typeOption
 	threadOpts *typeOption
+
+	pReportOpts *ReporterOptions
+}
+
+// rptEvent stands of the args of report event
+type rptEvent struct {
+	PType   string
+	Buf     []byte
+	Reason  string
+	EventID string
+}
+
+type ReporterOptions struct {
+	L        *sync.RWMutex
+	reporter ProfileReporter
+	active   int32 // switch
+	eventsCh chan rptEvent
+}
+
+// newReporterOpts starts a background goroutine to consume event channel
+func newReporterOpts() *ReporterOptions {
+	opts := &ReporterOptions{}
+	opts.eventsCh = make(chan rptEvent, 32)
+	opts.L = &sync.RWMutex{}
+
+	go func(eventsCh <-chan rptEvent) {
+		for {
+			evt := <-eventsCh
+			opts.reporter.Report(evt.PType, evt.Buf, evt.Reason, evt.EventID) // nolint: errcheck
+		}
+	}(opts.eventsCh)
```

Comment:
it's not a good idea to start the goroutine here. we should start it in the `Holmes.Start` method.
Also, we should respect the `Holmes.Stop` method.